exoctk/exoctk_app/app_exoctk.py

```diff
@@ -408,208 +409,79 @@ def contam_visibility():
         # Send it back to the main page
         return render_template('contam_visibility.html', form=form)
 
-    if form.validate_on_submit() and form.calculate_submit.data:
+    # Reload page with appropriate mode data
+    if form.mode_submit.data:
 
-        # Log the form inputs
-        try:
-            log_exoctk.log_form_input(request.form, 'contam_visibility', DB)
-        except:
-            pass
+        # Update the button
+        if form.inst.data != 'NIRISS':
+            form.calculate_contam_submit.disabled = True
+        else:
+            form.calculate_contam_submit.disabled = False
 
-        # contamVars = {}
-        # if request.method == 'POST':
-        #     tname = request.form['targetname']
-        #     contamVars['tname'] = tname
-        #     contamVars['ra'], contamVars['dec'] = request.form['ra'], request.form['dec']
-        #     contamVars['PAmax'] = request.form['pamax']
-        #     contamVars['PAmin'] = request.form['pamin']
-        #     contamVars['inst'] = request.form['inst'].split()[0]
-        #
-        #     if request.form['bininfo'] != '':
-        #         contamVars['binComp'] = list(map(float, request.form['bininfo'].split(', ')))
-        #     else:
-        #         contamVars['binComp'] = request.form['bininfo']
-        #
-        #     radec = ', '.join([contamVars['ra'], contamVars['dec']])
-        #
-        #     if contamVars['PAmax'] == '':
-        #         contamVars['PAmax'] = 359
-        #     if contamVars['PAmin'] == '':
-        #         contamVars['PAmin'] = 0
-        #
-        #     if request.form['submit'] == 'Resolve Target':
-        #         contamVars['ra'], contamVars['dec'] = resolve.resolve_target(tname)
-        #
-        #         return render_template('contam_visibility.html', contamVars=contamVars)
-        #
-        #     else:
+        # Send it back to the main page
+        return render_template('contam_visibility.html', form=form)
+
+    if form.validate_on_submit() and (form.calculate_submit.data or form.calculate_contam_submit.data):
 
         try:
 
+            # Log the form inputs
+            try:
+                log_exoctk.log_form_input(request.form, 'contam_visibility', DB)
+            except:
+                pass
+
             # Calculate
+            title = form.targname.data or ', '.join([form.ra.data, form.dec.data])
             pG, pB, dates, vis_plot, table = vpa.using_gtvt(str(form.ra.data), str(form.dec.data), form.inst.data)
```

Comment:
heh, this is actually returning `paMin, paMax, dates, vis_plot, table`. I'll go ahead and make a PR to fix this.

Reply:
Oh, are the badPA values just the NANs in the date array? I can just pull this out for the contemn plot if so.

Reply:
Oh, well it works as is. It just doesn't plot the bad PA values on the contamination plot. Let's fix this in another PR because this one is getting bloated.